index.html

```diff
@@ -3,47 +3,256 @@
+	{"state":"Wisconsin","employment":2936.2,"region":"Midwest"},
+	{"state":"Wyoming","employment":284.4,"region":"West"}
+  ];
+
+  var MAX_EMPLOYMENT = 16322;
+
+  var options = {
+    sorted: false,
+    filtered: false,
+    page: 0
+  };
+    
+    
+    var tbody = d3.select("tbody");
+
+  function updateTable(){
+    var rows = "";
+    tbody.html("");
+    var data = states.slice();
+      
+    if (options.sorted) {
+      if (options.sorted === "asc") {
+          data.sort(function (a, b) { return a.employment - b.employment; });
+      } else if (options.sorted === "desc") {
+          data.sort(function (a, b) { return b.employment - a.employment; });
+      } 
+    }        
+    
+
+    if (options.filtered) {
+      data = data.filter(function (d) { 
+          return d.region === options.filtered; 
+          });
+    }      
+            
+    rows = tbody.selectAll("tr")
+        .data(data)
+        .enter()
+        .append("tr");
+
+    rows.append("td")
+      .html(function(d) { return d.state });
+      
+    rows.append("td")
+        .html(function(d) { return d.employment });
+
+    rows.append("td")
+      .html(function(d) { return d.region });
+
+    rows.append("td")
+      .append("div")
+      .classed("bar", true)
+      .style("width", function(d) { return ((d.employment / MAX_EMPLOYMENT) * 100) + "%" })
+
+  };   
+    
+  updateTable();
+
+
+// sort ascending 
+    d3.select("#sort-asc").on("click", function() {
+        if (options.sorted === "asc") {
+        options.sorted = false;
+        d3.select("#sort-asc").classed("active", false);    
+        } else {
+            options.sorted = "asc";
+            d3.select("#sort-asc").classed("active", true);
+            d3.select("#sort-desc").classed("active", false);
+        }
+        updateTable();
+    });
+
+// sort descending    
+    d3.select("#sort-desc").on("click", function() {
+        if (options.sorted === "desc") {
+        options.sorted = false;
+        d3.select("#sort-desc").classed("active", false);    
+        } else {
+            options.sorted = "desc";
+            d3.select("#sort-desc").classed("active", true);
+            d3.select("#sort-asc").classed("active", false);
+        }
+        updateTable();
+    });    
+    
+// filter for the 4 regions
+    d3.select("#filter-ne").on("click", function() {
+        if (options.filtered === "Northeast") {
+            options.filtered = false;
+            d3.select("#filter-ne").classed("active", false);                
+        } else {
+            options.filtered = "Northeast";
+            d3.select("#filter-so").classed("active", false);                            
+            d3.select("#filter-mw").classed("active", false);                                        
+            d3.select("#filter-we").classed("active", false);                                        
+            d3.select("#filter-ne").classed("active", true);                            
```

Comment:
You're on the right track, but note that you're now setting the active class to the group, not the buttons...